src/math.rs

```diff
@@ -0,0 +1,47 @@
+//! Floating point arithmetic approximations for `no_std` targets
```

Comment:
These are vendored from the (i.e. my) `micromath` crate, where they all have tests:

https://github.com/NeoBirth/micromath/blob/develop/src/f32ext/abs.rs
https://github.com/NeoBirth/micromath/blob/develop/src/f32ext/floor.rs
https://github.com/NeoBirth/micromath/blob/develop/src/f32ext/cos.rs
https://github.com/NeoBirth/micromath/blob/develop/src/f32ext/sin.rs

If you'd like I can vendor the tests as well.